src/System.Private.CoreLib/shared/System/TimeZoneInfo.Unix.cs

```diff
@@ -478,6 +487,11 @@ private static IEnumerable<string> EnumerateFilesRecursively(string path)
                         }
                     }
                 }
+                finally
+                {
+                    ArrayPool<byte>.Shared.Return(dirBuffer);
```

Comment:
Check for null. You can end up here with null `dirBuffer` if `Rent` throws. `Return` will throw `ArgumentException` that would hide the  original error.